src/tribler/core/utilities/pony_utils.py

```diff
@@ -28,6 +34,87 @@
 StatDict = Dict[Optional[str], core.QueryStat]
 
 
+class DatabaseIsCorrupted(Exception):
+    pass
+
+
+def table_exists(cursor: sqlite3.Cursor, table_name: str) -> bool:
+    cursor.execute("SELECT name FROM sqlite_master WHERE type='table' AND name=?", (table_name,))
+    return cursor.fetchone() is not None
+
+
+def get_db_version(db_path, default: int = None) -> int:
+    handle_db_if_corrupted(db_path)
+    if not db_path.exists():
+        version = None
+    else:
+        with marking_corrupted_db(db_path):
+            with contextlib.closing(sqlite3.connect(db_path)) as connection:
+                with connection:
+                    cursor = connection.cursor()
+                    if not table_exists(cursor, 'MiscData'):
+                        version = None
+                    else:
+                        cursor.execute("SELECT value FROM MiscData WHERE name == 'db_version'")
+                        row = cursor.fetchone()
+                        version = int(row[0]) if row else None
+
+    if version is not None:
+        return version
+
+    if default is not None:
+        return default
+
+    raise RuntimeError(f'The version value is not found in database {db_path}')
+
+
+def handle_db_if_corrupted(db_filename: Union[str, Path]):
+    db_path = Path(db_filename)
+    marker_path = _get_corrupted_db_marker_path(db_path)
+    if marker_path.exists():
+        _handle_corrupted_db(db_path)
+
+
+def _handle_corrupted_db(db_path: Path):
+    if db_path.exists():
+        logger.warning(f'Database file was marked as corrupted, removing it: {db_path}')
+        db_path.unlink()
+
+    marker_path = _get_corrupted_db_marker_path(db_path)
+    if marker_path.exists():
+        logger.warning(f'Removing the corrupted database marker: {marker_path}')
+        marker_path.unlink()
```

Comment:
Should we handle potential `OSErrors` here, for instance, in cases where `unlink` raises an exception?

Reply:
In a normal situation, we should not have any `OSErrors` here. If we have them, it may be a sign that two processes run in parallel and create a race condition. In that case, we should crash with an error. Otherwise, two parallel Core processes can perform inconsistent changes in the database. If this type of error is possible, we should be brittle on it and not allow two Core processes to run in parallel.